src/main/public/js/analytics-custom-event-sender.js

```diff
@@ -0,0 +1,69 @@
+$(function () {
+  function sendEvent(eventCategory, eventAction, eventLabel) {
+    if (!eventCategory) {
+      throw new Error('Event category is required')
+    }
+    if (!eventAction) {
+      throw new Error('Event action is required')
+    }
+    if (!eventLabel) {
+      throw new Error('Event label is required')
+    }
+
+    ga('send', 'event', eventCategory, eventAction, eventLabel)
+  }
+
+  function labelText(labelElement) {
+    if (!labelElement) {
+      throw new Error('Label element is required')
+    }
+
+    for (var content of labelElement.contents()) {
+      const text = $(content).text().trim()
+      if (text && content.class !== 'form-hint') {
+        return text
+      }
+    }
+    return undefined
+  }
+
+  function formValue(formElement, inputName) {
```

Comment:
I think this name is a little unfortunate because it ultimately retrieves a label, not a value.

Reply:
I think I'd also use just `form` for first parameter (and update in other places accordingly). This confused me a little because I thought it's an element of a form, not the form itself. Not strong on it though.

Reply:
Fair point, I just used to call all DOM elements retrieved using jQuery as `xyzElement`. I am happy to change it as it is probably just mine convention.